### ISSUE TITLE
Support for unlocking encrypted Tox profiles using fingerprint/biometrics

### ISSUE THREAD

**Issue by robinlinden** [evilcorpltd/aTox#917]

**Comment by roihershberg** [comment]
It can expose your keys if the device integrity has been broken as can be seen [here](https://security.stackexchange.com/questions/124764/how-to-crack-get-secret-from-android-keystore-system).
If you have a rooted device or a device with hardware/firmware that is closed source (almost all of them) and who knows if it isn't wired in such a way that it can be easily dumped. I heard that every phone is wired in their modem to dump all the storage over the cellular network if they receive a specific command. They excuse it as allowing the police to investigate the phone's storage of suspects but it's essentially a doorway to access the storage of every single device every time they want to. So if they want to, they can dump that phone's storage, easily crack the encryption and access that keystore if it's software-backed. Or they can even dump the hardware-backed keystores as well if it's configured to do so. The device can also simply have an OS without basic integrity that sends that keystore to some entity.

But well, maybe we should still support it for those who wish to.

**Comment by roihershberg** [comment]
Is it duplicate of #736?

**Comment by robinlinden** [comment]
> Is it duplicate of #736?

No, that's more about using a hardware-protected profile, so even if it were dumped from the phone by e.g. the disk being cloned, it wouldn't be usable.
